gpbft/gpbft.go

```diff
@@ -544,6 +548,12 @@ func (i *instance) tryConverge() error {
 		return nil
 	}
 
+	// Check strong quorum values from late arriving QUALITY messages and update
+	// candidates accordingly.
+	if foundQualityQuorum := i.quality.HasStrongQuorumFor(i.initialProposal.Key()); foundQualityQuorum {
+		i.addCandidatePrefixes(i.initialProposal)
+	}
```

Comment:
We need to walk all prefixes of `initialProposal` and call `addCandidatePrefixes` on the first prefix with strong quorum.